src/mono/mono/mini/mini-arm64.c

```diff
@@ -1497,16 +1498,30 @@ is_hfa (MonoType *t, int *out_nfields, int *out_esize, int *field_offsets)
 			int nested_nfields, nested_esize;
 			int nested_field_offsets [16];
 
-			if (!is_hfa (ftype, &nested_nfields, &nested_esize, nested_field_offsets))
-				return FALSE;
+			MonoType *fixed_etype;
+			int fixed_len;
+			if (mono_marshal_shared_get_fixed_buffer_attr (field, &fixed_etype, &fixed_len)) {
+				if (fixed_etype->type != MONO_TYPE_R4 && fixed_etype->type != MONO_TYPE_R8)
+					return FALSE;
+				if (fixed_len > 16)
```

Comment:
for it to be an HFA doesn't it need to be 2-4 items, so we need to make sure fixed_len >= (sizeof (etype) * 2)? not that a fixed array of size 1 makes sense but you could do it

Reply:
EDIT: I see this is bounds checking against nested_field_offsets now, not against 16 / sizeof(etype)